crates/cxx-qt-gen/src/parser/signals.rs

```diff
@@ -26,27 +27,42 @@ pub struct ParsedSignal {
     pub cfgs: Vec<Attribute>,
 }
 
+pub enum MutabilityConstraint {
+    Allowed,
+    Disallowed,
+}
+
 impl ParsedSignal {
     const ALLOWED_ATTRS: [&'static str; 6] =
         ["cfg", "cxx_name", "rust_name", "inherit", "doc", "qsignal"];
 
     #[cfg(test)]
     /// Test fn for creating a mocked signal from a method body
     pub fn mock(method: &ForeignItemFn) -> Self {
-        Self::parse(method.clone(), CaseConversion::none()).unwrap()
+        Self::parse_rust_qt_signal(method.clone(), CaseConversion::none()).unwrap()
     }
 
-    pub fn parse(method: ForeignItemFn, auto_case: CaseConversion) -> Result<Self> {
+    pub fn parse_with_mutability(
+        method: ForeignItemFn,
+        auto_case: CaseConversion,
+        mutability_allowed: MutabilityConstraint,
+    ) -> Result<Self> {
         let docs = extract_docs(&method.attrs);
         let cfgs = extract_cfgs(&method.attrs);
         let fields = MethodFields::parse(method, auto_case)?;
         let attrs = require_attributes(&fields.method.attrs, &Self::ALLOWED_ATTRS)?;
 
-        if !fields.mutable {
-            return Err(Error::new(
-                fields.method.span(),
-                "signals must be mutable, use Pin<&mut T> instead of T for the self type",
-            ));
+        match mutability_allowed {
```

Comment:
could potentially have one if statement?

```rust
if matches!(mutability_allowed, MutabilityContraint::Disallowed) && !fields.mutable {
...
}
```

Although this then reads curiously should it be `ImmutabilityConstraint` ? As when the constraint is disallowed we are not allowed to have immutability but we are allowed to have mutability. Whereas when you read this is sounds a little like the opposite?